subsys/net/lib/lwm2m/lwm2m_rw_plain_text.c

```diff
@@ -334,6 +333,64 @@ const struct lwm2m_reader plain_text_reader = {
 	get_opaque
 };
 
+int do_read_op_plain_text(struct lwm2m_engine_obj *obj,
+			  struct lwm2m_engine_context *context,
+			  int content_format)
+{
+	struct lwm2m_obj_path *path = context->path;
+	struct lwm2m_engine_obj_inst *obj_inst = NULL;
+	struct lwm2m_engine_res_inst *res = NULL;
+	struct lwm2m_engine_obj_field *obj_field;
+	int ret = 0, index;
+
+	/* Plain text can only return single resource */
+	if (path->level >= 3) {
```

Comment:
How about this instead here:
```
if (path->level == 3) {
```

Reply:
I would assume that this is correct but I've no idea what exactly `level` is expressing.

Reply:
"level" is the depth of the uri.  Since we're using unsigned integer types and a value of "0" is valid, we need a way of knowing how much of the URI was set.

IE:
/3 == level 1, with values: obj_id=3, obj_inst_id=0, res_id=0
/3/0 == level 2, with values: obj_id=3, obj_inst_id=0, res_id=0
/3/0/0 == level 3, with values: obj_id=3, obj_inst_id=0, res_id=0

Reply:
That matches with my assumption so `== 3` would be correct. The only thing I'm not clear about is whether a level == 4 might be used to express multiple resources. If so then `>= 3` would be wrong because multiple resources can not be expressed in plain text. If level == 4 cannot happen anyways then then point is moot. ;)